File: 9-PalindromeNumber/9-PalindromeNumber.js
// Last updated: 24/07/2026, 23:55:38
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let isNegative = x < 0;
    if(isNegative) return false;

    let rev = 0;
    let tmp = x
    while(tmp > 0){
        rev = ((tmp % 10)) + rev * 10
        // console.log(rev)
        tmp = Math.floor(tmp/10);
    }

    return rev == x
};

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 24/07/2026, 23:55:44
class Solution:
    def isPalindrome(self, x: int) -> bool:
        n2=x
        rev=0
        while x>0:
            n1=x%10
            rev = rev*10+n1
            x=x//10
        if n2==rev:
            return True
        else:
            return False